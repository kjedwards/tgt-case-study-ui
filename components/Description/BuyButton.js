/* ----------------------------------------------
 * Component for Pick Up and Cart buttons
 *
 * Author: Kaylee Edwards (z0027hd)
 * Last Modified: 06/08/2018
 * Last Modified By: Kaylee Edwards (z0027hd)
 * --------------------------------------------*/

import React, {Component} from 'react';
import styled, {css} from 'styled-components'

const Container = styled.div`
  display: flex;
  flex-direction: row;
  text-align: center;
  margin-right: 5px;
  width: 100%;
`

const Pickup = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: center;
  flex-grow: 1;
`

const PickupButton = styled.button`
  background-color: #3a3a3a;
  padding: 5px 20px;
  color: white;
  border-radius: 3px;
  border: 0px;
  height: 30px;
`

const CartButton = styled.button`
  background-color: #e50909;
  padding: 5px 20px;
  color: white;
  border-radius: 3px;
  border: 0px;
  height: 30px;
  flex-grow: 1;
`

const FindStore = styled.a`
  color: black;
  font-size: 10px;
  font-weight: bold;
  padding: 3px;
`

class BuyButton extends Component {
    render () {
      return(
        <Container>
          { this.props.render ?
              this.props.type == 'pickup' ?
                <Pickup>
                  <PickupButton>PICK UP IN STORE</PickupButton>
                  <FindStore>find in a store</FindStore>
                </Pickup>
                : <CartButton>ADD TO CART</CartButton>
            : null
          }

        </Container>
      )
    }
}

export default BuyButton
