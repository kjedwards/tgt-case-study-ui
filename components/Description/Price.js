/* ----------------------------------------------
 * Component displaying the price
 *
 * Author: Kaylee Edwards (z0027hd)
 * Last Modified: 06/08/2018
 * Last Modified By: Kaylee Edwards (z0027hd)
 * --------------------------------------------*/

import React, {Component} from 'react';
import styled, {css} from 'styled-components'

const Container = styled.div`
  width: 100%;
  margin-bottom: 20px;
  display: flex;
`

const Number = styled.div`
  font-size: 24px;
  font-weight: bold;
`

const Subtext = styled.div`
  margin-top: 16px;
  margin-left: 3px;
  font-size: 8px;

`


class Price extends Component {
    render () {
      return(
        <Container>
          <Number>{this.props.offer.formattedPriceValue}</Number>
          <Subtext>{this.props.offer.priceQualifier.toLowerCase()}</Subtext>
        </Container>
      )
    }
}


export default Price
