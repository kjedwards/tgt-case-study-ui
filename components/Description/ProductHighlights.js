/* ----------------------------------------------
 * Component displaying the product highlights section
 *
 * Author: Kaylee Edwards (z0027hd)
 * Last Modified: 06/08/2018
 * Last Modified By: Kaylee Edwards (z0027hd)
 * --------------------------------------------*/

import React, {Component} from 'react';
import styled, {css} from 'styled-components'

const Container = styled.div`
  display: flex;
  flex-direction: column;
  margin-top: 15px;
`

const Header = styled.div`
  font-size: 32px;
  margin: 10px 0px;
`

const List = styled.ul`
  padding-left: 20px;
`

const ListMember = styled.li`
  padding: 3px;
`

class ProductHighlights extends Component {
    render () {
      return(
        <Container>
          <Header>product highlights</Header>
          <List>
            <ListMember>Wattage Output: 1100 watts</ListMember>
            <ListMember>Wattage Output: 1100 watts</ListMember>
            <ListMember>Wattage Output: 1100 watts</ListMember>
            <ListMember>Wattage Output: 1100 watts</ListMember>
            <ListMember>Wattage Output: 1100 watts</ListMember>
            <ListMember>Wattage Output: 1100 watts</ListMember>
            <ListMember>Wattage Output: 1100 watts</ListMember>
          </List>
        </Container>
      )
    }
}

export default ProductHighlights
