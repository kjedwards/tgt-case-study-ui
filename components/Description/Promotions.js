/* ----------------------------------------------
 * Component displaying the promotions
 *
 * Author: Kaylee Edwards (z0027hd)
 * Last Modified: 06/08/2018
 * Last Modified By: Kaylee Edwards (z0027hd)
 * --------------------------------------------*/

import React, {Component} from 'react';
import styled, {css} from 'styled-components'

import '../../node_modules/font-awesome/css/font-awesome.min.css'

const Container = styled.div`
  width: 100%;
  border-top: 1px solid grey;
  border-bottom: 1px solid grey;
  color: red;
  display: flex;
  flex-direction: column;
  padding-top: 10px;
  padding-bottom: 10px;
`

const Promotion = styled.div`
  display: flex;
  flex-direction: row;
  align-items: center;
  padding-top: 2px;
  padding-bottom: 2px;
`

const Tag = styled.div`
  font-size: 10px;
  padding-right: 5px;
`

const Text = styled.div`
  font-size: 12px;
`


class Promotions extends Component {
    render () {
      return(
        <Container>
          <Promotion>
            <Tag><i className='fa fa-tag'/></Tag>
            <Text>Wow, what a deal!</Text>
          </Promotion>
          <Promotion>
            <Tag><i className='fa fa-tag'/></Tag>
            <Text>Wow, what a deal!</Text>
          </Promotion>
        </Container>
      )
    }
}


export default Promotions
