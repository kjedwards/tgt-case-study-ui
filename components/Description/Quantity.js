/* ----------------------------------------------
 * Component displaying the quantity of items
 *
 * Author: Kaylee Edwards (z0027hd)
 * Last Modified: 06/08/2018
 * Last Modified By: Kaylee Edwards (z0027hd)
 * --------------------------------------------*/

import React, {Component} from 'react';
import styled, {css} from 'styled-components'

const Container = styled.div`
  margin-top: 15px;
  margin-bottom: 15px;
  border: 2px solid lightgrey;
  width: 150px;
  align-items: center;
`

const Text = styled.div`
  font-size: 11px;
  float: left;
  padding: 7px;
`

const PlusMinusContainer = styled.div`
  float: right;
  display: flex;
  align-items: center;
`

const Button = styled.div`
  color: lightgrey;
  padding: 3px 7px 2px 7px;
  font-size: 20px;
`

const Amount = styled.div`

`


class Quantity extends Component {
    render () {
      return(
        <Container>
          <Text>quantity:</Text>
          <PlusMinusContainer>
            <Button>
              <i className='fa fa-plus-circle' />
            </Button>
            <Amount> 1 </Amount>
            <Button>
              <i className='fa fa-minus-circle' />
            </Button>
          </PlusMinusContainer>
        </Container>
      )
    }
}


export default Quantity
