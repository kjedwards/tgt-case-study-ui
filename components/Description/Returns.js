/* ----------------------------------------------
 * Component to display the return policy
 *
 * Author: Kaylee Edwards (z0027hd)
 * Last Modified: 06/08/2018
 * Last Modified By: Kaylee Edwards (z0027hd)
 * --------------------------------------------*/

import React, {Component} from 'react';
import styled, {css} from 'styled-components'

const Container = styled.div`
  display: flex;
  align-items: center;
  margin: 10px;
`

const LabelText = styled.div`
  font-size: 16px;
  padding-right: 15px;
  border-right: 1px solid grey;
`

const DescText = styled.div`
  font-size: 8px;
  padding-left: 15px;
`

class Returns extends Component {
    render () {
      return(
        <Container>
          <LabelText>returns</LabelText>
          <DescText>
            This item must be returned within 30 days of
            the ship date. See <b>return policy</b> for details.
            Prices, promotions, styles, and availability may
            vary by store and online.
          </DescText>
        </Container>
      )
    }
}

export default Returns
