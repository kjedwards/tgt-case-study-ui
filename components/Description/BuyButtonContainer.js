import React, {Component} from 'react';
import styled, {css} from 'styled-components'

import BuyButton from './BuyButton'

const Container = styled.div`
  display: flex
  margin: 5px 0px;
`


class BuyButtonContainer extends Component {
    render () {
      return(
        <Container>
          <BuyButton></BuyButton>
          <BuyButton></BuyButton>
        </Container>
      )
    }
}

export default BuyButtonContainer
