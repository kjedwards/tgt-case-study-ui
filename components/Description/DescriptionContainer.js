/* ----------------------------------------------
 * Container for the product information
 *
 * Author: Kaylee Edwards (z0027hd)
 * Last Modified: 06/08/2018
 * Last Modified By: Kaylee Edwards (z0027hd)
 * --------------------------------------------*/

import React, {Component} from 'react';
import styled, {css} from 'styled-components'

import { SIZES } from '../../constants/DeviceWidth'

import Price from './Price'
import Promotions from './Promotions'
import Quantity from './Quantity'
import BuyButtonContainer from './BuyButtonContainer'
import Returns from './Returns'
import ShareContainer from './ShareContainer'
import ProductHighlights from './ProductHighlights'

const DescContainer = styled.div`
  order: 2;
  display: flex;
  flex-direction: column;
  max-width: 300px;

  @media(max-width: ${SIZES.TABLET}) {
    order: 1;
  }
`


class DescriptionContainer extends Component {
    render () {
      return(
        <DescContainer>
          <Price offer={this.props.offer}></Price>
          <Promotions promos={this.props.promo}></Promotions>
          <Quantity></Quantity>
          <BuyButtonContainer channelCode={this.props.channelCode}></BuyButtonContainer>
          <Returns></Returns>
          <ShareContainer></ShareContainer>
          <ProductHighlights highlights={this.props.highlights}></ProductHighlights>
        </DescContainer>
      )
    }
}


export default DescriptionContainer
