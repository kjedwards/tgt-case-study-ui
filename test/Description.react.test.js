import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';

//Imported Components
import BuyButton from '../components/Description/BuyButton'
import BuyButtonContainer from '../components/Description/BuyButtonContainer'
import DescriptionContainer from '../components/Description/DescriptionContainer'
import Price from '../components/Description/Price'
import ProductHighlights from '../components/Description/ProductHighlights'
import Promotions from '../components/Description/Promotions'
import Quantity from '../components/Description/Quantity'
import Returns from '../components/Description/Returns'
import ShareButton from '../components/Description/ShareButton'
import ShareContainer from '../components/Description/ShareContainer'

//Mocks
jest.mock('../components/Description/BuyButton', () => 'buy-button')
jest.mock('../components/Description/Price', () => 'price')
jest.mock('../components/Description/ProductHighlights', () => 'prouct-highlights')
jest.mock('../components/Description/Promotions', () => 'promotions')
jest.mock('../components/Description/Quantity', () => 'quantity')
jest.mock('../components/Description/Returns', () => 'returns')
jest.mock('../components/Description/ShareButton', () => 'share-button')

describe('BuyButton Component', () => {
  it('RENDERER snapshot', () => {
    const tree = renderer.create(<BuyButton />).toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('ENZYME snapshot', () => {
    const tree = mount(<BuyButton />)
    expect(toJson(tree)).toMatchSnapshot()
  })
})

describe('BuyButtonContainer Component', () => {
  it('RENDERER snapshot', () => {
    const tree = renderer.create(<BuyButtonContainer />).toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('ENZYME snapshot', () => {
    const tree = mount(<BuyButtonContainer />)
    expect(toJson(tree)).toMatchSnapshot()
  })
})

describe('Price Component', () => {
  it('RENDERER snapshot', () => {
    const tree = renderer.create(<Price />).toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('ENZYME snapshot', () => {
    const tree = mount(<Price />)
    expect(toJson(tree)).toMatchSnapshot()
  })
})

describe('Product Highlights Component', () => {
  it('RENDERER snapshot', () => {
    const tree = renderer.create(<ProductHighlights />).toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('ENZYME snapshot', () => {
    const tree = mount(<ProductHighlights />)
    expect(toJson(tree)).toMatchSnapshot()
  })
})

describe('Promotions Component', () => {
  it('RENDERER snapshot', () => {
    const tree = renderer.create(<Promotions />).toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('ENZYME snapshot', () => {
    const tree = mount(<Promotions />)
    expect(toJson(tree)).toMatchSnapshot()
  })
})

describe('Quantity Component', () => {
  it('RENDERER snapshot', () => {
    const tree = renderer.create(<Quantity />).toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('ENZYME snapshot', () => {
    const tree = mount(<Quantity />)
    expect(toJson(tree)).toMatchSnapshot()
  })
})

describe('Returns Component', () => {
  it('RENDERER snapshot', () => {
    const tree = renderer.create(<Returns />).toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('ENZYME snapshot', () => {
    const tree = mount(<Returns />)
    expect(toJson(tree)).toMatchSnapshot()
  })
})

describe('ShareButton Component', () => {
  it('RENDERER snapshot', () => {
    const tree = renderer.create(<ShareButton />).toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('ENZYME snapshot', () => {
    const tree = mount(<ShareButton />)
    expect(toJson(tree)).toMatchSnapshot()
  })
})

describe('ShareContainer Component', () => {
  it('RENDERER snapshot', () => {
    const tree = renderer.create(<ShareContainer />).toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('ENZYME snapshot', () => {
    const tree = mount(<ShareContainer />)
    expect(toJson(tree)).toMatchSnapshot()
  })
})
