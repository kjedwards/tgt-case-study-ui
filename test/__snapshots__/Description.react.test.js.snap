// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BuyButton Component ENZYME snapshot 1`] = `<buy-button />`;

exports[`BuyButton Component RENDERER snapshot 1`] = `<buy-button />`;

exports[`BuyButtonContainer Component ENZYME snapshot 1`] = `
<BuyButtonContainer>
  <styled.div>
    <div
      className="sc-bdVaJa gkBsxR"
    >
      <buy-button
        render={false}
        type="pickup"
      />
      <buy-button
        render={false}
        type="cart"
      />
      <p>
        Item Unavailable
      </p>
    </div>
  </styled.div>
</BuyButtonContainer>
`;

exports[`BuyButtonContainer Component RENDERER snapshot 1`] = `
<div
  className="sc-bdVaJa gkBsxR"
>
  <buy-button
    render={false}
    type="pickup"
  />
  <buy-button
    render={false}
    type="cart"
  />
  <p>
    Item Unavailable
  </p>
</div>
`;

exports[`Price Component ENZYME snapshot 1`] = `<price />`;

exports[`Price Component RENDERER snapshot 1`] = `<price />`;

exports[`Product Highlights Component ENZYME snapshot 1`] = `<prouct-highlights />`;

exports[`Product Highlights Component RENDERER snapshot 1`] = `<prouct-highlights />`;

exports[`Promotions Component ENZYME snapshot 1`] = `<promotions />`;

exports[`Promotions Component RENDERER snapshot 1`] = `<promotions />`;

exports[`Quantity Component ENZYME snapshot 1`] = `<quantity />`;

exports[`Quantity Component RENDERER snapshot 1`] = `<quantity />`;

exports[`Returns Component ENZYME snapshot 1`] = `<returns />`;

exports[`Returns Component RENDERER snapshot 1`] = `<returns />`;

exports[`ShareButton Component ENZYME snapshot 1`] = `<share-button />`;

exports[`ShareButton Component RENDERER snapshot 1`] = `<share-button />`;

exports[`ShareContainer Component ENZYME snapshot 1`] = `
<ShareContainer>
  <styled.div>
    <div
      className="sc-bwzfXH fIZIDy"
    >
      <share-button
        text="ADD TO REGISTRY"
      />
      <share-button
        text="ADD TO LIST"
      />
      <share-button
        text="SHARE"
      />
    </div>
  </styled.div>
</ShareContainer>
`;

exports[`ShareContainer Component RENDERER snapshot 1`] = `
<div
  className="sc-bwzfXH fIZIDy"
>
  <share-button
    text="ADD TO REGISTRY"
  />
  <share-button
    text="ADD TO LIST"
  />
  <share-button
    text="SHARE"
  />
</div>
`;
